### ISSUE TITLE
Thank you for working on this extension

### ISSUE THREAD

**Issue by vemel** [chrishollinworth/vscode-boto3-intellisense#1]
Hello,

I have noticed this repo and I want to thank you! I attempted to create a vscode extensions that does this:

- Grep for `boto3.session/client/resource` calls in repo and extract service names from them
- Install `mypy-boto3-*` libraries according to fond service names
- Generate `typings/boto3-stubs` package with overloads for installed `mypy-boto3-*` libraries

However, I stuck on the second step as I do not know how to `pip install` properly from vscode extension.

Let me know if you need any help, I probably still have a source code to share.

Regards, Vlad.

**Comment by chrishollinworth** [comment]
HI Vlad,

Thanks for your wonderful mypy_boto3_builder code !, it makes generating type stubs so much easier.

I decided to use pyright/pylance extension as they use type stubs out of the box automatically.

Making a code completion vscode execution requires typescript and language servers, so I decided to leverage pyright/pylance functionality and I'm really impressed with the type inference if offers.

if you want to run shell commands from inside a vscode extension, take a look here https://stackoverflow.com/questions/43007267/how-to-run-a-system-command-from-vscode-extension/43008075

It probably will need OS detection to run per OS commands.

Give me a shout, if you need a hand.

Regards,
Chris

**Comment by vemel** [comment]
`Python` extension has a proper implementation of `pip install`, but it is too complex and I did not want to just copy-paste it.

I am also using Pyright currently. They fixed the issue I reported for n^2 complexity for function overloads in the latest release: https://github.com/microsoft/pyright/releases/tag/1.1.109 . So, it should be even faster now, and there is no need to generate `boto3-stubs` locally.
